test/e2e/windows/hybrid_network.go

```diff
@@ -0,0 +1,133 @@
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	linuxOS   = "linux"
+	windowsOS = "windows"
+)
+
+var (
+	windowsBusyBoximage = imageutils.GetE2EImage(imageutils.TestWebserver)
+	linuxBusyBoxImage   = "docker.io/library/nginx:1.15-alpine"
+)
+
+var _ = SIGDescribe("Hybrid cluster network", func() {
+	f := framework.NewDefaultFramework("hybrid-network")
+
+	BeforeEach(func() {
+		framework.SkipUnlessNodeOSDistroIs("windows")
+	})
+
+	Context("for all supported CNIs", func() {
+
+		It("should have stable networking for Linux and Windows pods", func() {
+			By("creating linux and windows pods")
+			linuxPod := createTestPod(f, linuxBusyBoxImage, linuxOS)
+			windowsPod := createTestPod(f, windowsBusyBoximage, windowsOS)
+
+			By("checking connectivity to 8.8.8.8 53 (google.com) from Linux")
+			assertConsistentConnectivity(f, linuxPod.ObjectMeta.Name, linuxOS, linuxCheck("8.8.8.8", 53))
+
+			By("checking connectivity to www.google.com from Windows")
+			assertConsistentConnectivity(f, windowsPod.ObjectMeta.Name, windowsOS, windowsCheck("www.google.com"))
+
+			By("checking connectivity from Linux to Windows")
+			assertConsistentConnectivity(f, linuxPod.ObjectMeta.Name, linuxOS, linuxCheck(windowsPod.Status.PodIP, 80))
+
+			By("checking connectivity from Windows to Linux")
+			assertConsistentConnectivity(f, windowsPod.ObjectMeta.Name, windowsOS, windowsCheck(linuxPod.Status.PodIP))
+
+		})
+
+	})
+})
+
+var (
+	duration      = "10s"
+	poll_interval = "1s"
+	timeout       = 10 // seconds
+)
+
+func assertConsistentConnectivity(f *framework.Framework, podName string, os string, cmd []string) {
+	Consistently(func() error {
+		By(fmt.Sprintf("checking connectivity of %s-container in %s", os, podName))
+		_, _, err := f.ExecCommandInContainerWithFullOutput(podName, os+"-container", cmd...)
+		return err
+	}, duration, poll_interval).ShouldNot(HaveOccurred())
+}
+
+func linuxCheck(address string, port int) []string {
+	nc := fmt.Sprintf("nc -vz %s %v -w %v", address, port, timeout)
+	cmd := []string{"/bin/sh", "-c", nc}
+	return cmd
+}
+
+func windowsCheck(address string) []string {
+	curl := fmt.Sprintf("curl.exe %s --connect-timeout %v --fail", address, timeout)
+	cmd := []string{"cmd", "/c", curl}
+	return cmd
+}
+
+func createTestPod(f *framework.Framework, image string, os string) *v1.Pod {
+	containerName := fmt.Sprintf("%s-container", os)
+	podName := "pod-" + string(uuid.NewUUID())
+	pod := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: podName,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  containerName,
+					Image: image,
+					Ports: []v1.ContainerPort{{ContainerPort: 80}},
+				},
+			},
+			NodeSelector: map[string]string{
+				"beta.kubernetes.io/os": os,
+			},
+		},
+	}
+	if os == linuxOS {
+		pod.Spec.Tolerations = []v1.Toleration{
+			{
+				Key:      "key",
```

Comment:
I think this is right that we need a toleration for a Linux taint, but as far as I know there's no standard name for this taint today. We use a key of "linux" and a dummy value for ours. I don't know what the aks-engine team uses.

Reply:
I spun up an aks-engine cluster to see the master had a `key=value:NoSchedule' taint so that's what I went off of... maybe @PatrickLang could help verify?

Reply:
I've removed the dependency to have key name